python/common/org/python/types/Str.java

```diff
@@ -1190,10 +1190,46 @@ private long toPositiveIndex(long index) {
                     "\n" +
                     "Search for the separator sep in S, starting at the end of S, and return\n" +
                     "the part before it, the separator itself, and the part after it.  If the\n" +
-                    "separator is not found, return two empty strings and S.\n"
+                    "separator is not found, return two empty strings and S.\n",
+            args = {"sep"}
     )
-    public org.python.Object rpartition() {
-        throw new org.python.exceptions.NotImplementedError("rpartition() has not been implemented.");
+    public org.python.Object rpartition(org.python.types.Object sep) {
+        java.lang.String sepStr = ((org.python.types.Str) sep).value;
+        java.lang.String str_one = new java.lang.String();
+        java.util.List<org.python.Object> tuple = new java.util.ArrayList<org.python.Object>();
+        if (sepStr.equals("")) {
+            throw new org.python.exceptions.ValueError("empty separator");
+        }
+        java.lang.String[] split = this.value.split(sepStr);
```

Comment:
What happens if you pass in some separator string that can be interpreted as a regex? `java.lang.String.split()` will interpret it as a regex, but will python's `Str.rpartition()`?

If it shouldn't be interpreted as a regex, could you use something like `java.lang.String.lastIndexOf()` instead?